python/graphstorm/model/edge_gnn.py

```diff
@@ -22,9 +22,10 @@
 import dgl
 
 from .gnn import GSgnnModel, GSgnnModelBase
+from .gnn_encoder_base import fix_for_wholegraph
 from .utils import append_to_dict
 
-from ..utils import barrier, is_distributed, get_rank
+from ..utils import barrier, is_distributed, get_rank, use_wholegraph
```

Comment:
use_wholegraph is slow as it will access local dir every time.
You can maintain a global variable for that.

Reply:
Fixed